boa/src/builtins/regexp/mod.rs

```diff
@@ -985,7 +985,7 @@ impl RegExp {
         let named_groups = match_value.named_groups();
         let groups = if named_groups.clone().count() > 0 {
             // a. Let groups be ! OrdinaryObjectCreate(null).
-            let groups = JsValue::new_object(context);
+            let groups = JsValue::from(Object::create(JsValue::null()));
```

Comment:
```suggestion
            let groups = JsObject::new(Object::create(JsValue::null()));
```

To avoid the `to_object` that is immediately executed in the for loop.

Maybe we should consider removing the explicit creation of `Object`s and just adding functions to create `JsObject`s. They're already coupled as a single unit, so I don't see the utility in having to create them separately.

Reply:
I remember that the default creation used the default object prototype to inherit from, but this one needed to inherit from a null prototype. But your suggestion makes full sense.